app/src/main/java/mozilla/lockbox/view/FilterFragment.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.view
+
+import android.content.Context
+import android.os.Bundle
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import com.jakewharton.rxbinding2.view.clicks
+import com.jakewharton.rxbinding2.view.visibility
+import com.jakewharton.rxbinding2.widget.text
+import com.jakewharton.rxbinding2.widget.textChanges
+import io.reactivex.Observable
+import io.reactivex.functions.Consumer
+import kotlinx.android.synthetic.main.fragment_filter.view.*
+import kotlinx.android.synthetic.main.include_backable_filter.view.*
+import mozilla.lockbox.R
+import mozilla.lockbox.adapter.ItemListAdapter
+import mozilla.lockbox.model.ItemViewModel
+import mozilla.lockbox.presenter.FilterPresenter
+import mozilla.lockbox.presenter.FilterView
+
+class FilterFragment : BackableFragment(), FilterView {
+    val adapter = ItemListAdapter()
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        presenter = FilterPresenter(this)
+        val view = inflater.inflate(R.layout.fragment_filter, container, false)
+        setupBackable(view, R.drawable.ic_arrow_back)
+
+        val layoutManager = LinearLayoutManager(context)
+        view.entriesView.layoutManager = layoutManager
+        view.entriesView.adapter = adapter
+        val decoration = DividerItemDecoration(context, layoutManager.orientation)
+        val decorationDrawable = context?.getDrawable(R.drawable.inset_divider)
+        decorationDrawable?.let { decoration.setDrawable(it) }
+        view.entriesView.addItemDecoration(decoration)
+
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        view!!.filterField.requestFocus()
+
+        val imm = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.showSoftInput(view!!.filterField, InputMethodManager.SHOW_IMPLICIT)
+    }
+
+    override val filterTextEntered: Observable<CharSequence> = view!!.filterField.textChanges()
```

Comment:
I'm getting a stack trace from line when I tap on the magnifying glass. 

```
     Caused by: kotlin.KotlinNullPointerException
        at mozilla.lockbox.view.FilterFragment.<init>(FilterFragment.kt:61)
        at mozilla.lockbox.presenter.RoutePresenter$filter$2.invoke(RoutePresenter.kt:32)
        at mozilla.lockbox.presenter.RoutePresenter$filter$2.invoke(RoutePresenter.kt:25)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at mozilla.lockbox.presenter.RoutePresenter.getFilter(Unknown Source:26)
        at mozilla.lockbox.presenter.RoutePresenter.route(RoutePresenter.kt:72)
        at mozilla.lockbox.presenter.RoutePresenter.access$route(RoutePresenter.kt:25)
        at mozilla.lockbox.presenter.RoutePresenter$1.invoke(RoutePresenter.kt:36)
        at mozilla.lockbox.presenter.RoutePresenter$1.invoke(RoutePresenter.kt:25)
        at mozilla.lockbox.presenter.RoutePresenter$sam$io_reactivex_functions_Consumer$0.accept(Unknown Source:21)
```

Reply:
You should also add test into the `RoutePresenterTest` and `Navigator`; this would've caught this problem.